include/systems/implicit_system.h

```diff
@@ -352,6 +357,11 @@ class ImplicitSystem : public ExplicitSystem
   bool zero_out_matrix_and_rhs;
 
 
+  /**
+   * Initializes the matrices associated with this system.
+   */
+  virtual void init_matrices ();
```

Comment:
I thought I would need this to be public. But after working on moose after this, turns out I only need to do
```
  // do not assemble system matrix for JFNK solve
  if (solverParams()._type == Moose::ST_JFNK)
    _nl->system().set_basic_system_only();
  _nl->nonlinearSolver()->jacobian = NULL;
```
in MOOSE `FEProblemBase::init`. So this does not have to be public to me. However, it may be desired in general.